### PR TITLE
Fix/fix set place rating qa remarks

### PR DESCRIPTION
**Pull request by fadykstas**
merge #301 first

### CODE REVIEW

File: app/Actions/Place/GetPlaceItem/GetPlaceItemPresenter.php [hunk 1]
```diff
@@ -61,6 +61,7 @@ public function present(GetPlaceItemResponse $placeResponse): array
         $result['localization'] = $this->localizationPresenter->presentCollection($place->localization);
         $result['category'] = $this->categoryPresenter->present($place->category);
         $result['category']['tags'] = $this->tagsPresenter->presentCollection($place->category->tags);
+        $result['myRating'] = (int)$place->ratings->where('user_id', $placeResponse->getUser()->id)->avg('rating');
```
Comment (fadykstas) [inline app/Actions/Place/GetPlaceItem/GetPlaceItemPresenter.php]: not sure if the right place

Comment (nemoipaha) [reply app/Actions/Place/GetPlaceItem/GetPlaceItemPresenter.php]: We have method for that in ratingRepository

File: app/Actions/Presenters/Place/PlacePresenter.php [hunk 2]
```diff
@@ -16,6 +16,7 @@ public function present(Place $place)
             'dislikes' => $place->dislikes->count(),
             'likes' => $place->likes->count(),
             'rating' => number_format(round($place->ratings->avg('rating'), 1), 1),
+            'ratingCount' => $place->ratings->count('rating'),
```
Comment (nemoipaha) [inline app/Actions/Presenters/Place/PlacePresenter.php]: just count() should also work

File: app/Http/Controllers/Api/Place/PlaceRatingController.php [hunk 3]
```diff
@@ -60,36 +60,20 @@ public function setRating(SetRatingHttpRequest $httpRequest) : JsonResponse
             'id' => $setPlaceRatingResponse->getId(),
             'user_id' => $setPlaceRatingResponse->getUserId(),
             'place_id' => $setPlaceRatingResponse->getPlaceId(),
-            'rating' => $setPlaceRatingResponse->getRatingValue(),
+            'my_rating' => $setPlaceRatingResponse->getRatingValue(),
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Place/PlaceRatingController.php]: Why we need my_rating here ?

File: database/migrations/2018_08_06_065718_create_place_rating_table.php [hunk 4]
```diff
@@ -20,7 +20,7 @@ public function up()
             $table->increments('id');
             $table->unsignedInteger('user_id');
             $table->unsignedInteger('place_id');
-            $table->unsignedDecimal('rating', 8, 2);
+            $table->unsignedInteger('rating');
```
Comment (nemoipaha) [inline database/migrations/2018_08_06_065718_create_place_rating_table.php]: no need to customize type of this field

File: frontend/src/components/place/PlaceCheckinModal.vue [hunk 5]
```diff
@@ -11,12 +11,14 @@
...
                         :icon="icon"
+                        :selected="isSelected(index + 1)"
                         @onHover="onHover"
+                        @onOut="onOut"
                         @onSelect="onSelect"
                     />
                 </div>
 
-                <span class="rating">{{ userRating }}/10</span>
+                <span class="rating">{{ userRating || place.myRating || 0 }}/10</span>
```
Comment (nemoipaha) [inline frontend/src/components/place/PlaceCheckinModal.vue]: not sure if we need myRating here

File: app/Http/Controllers/Api/Place/PlaceRatingController.php (hunk 3 above)
Comment (fadykstas) [reply app/Http/Controllers/Api/Place/PlaceRatingController.php]: `my_rating` is clearer indication than `rating`

File: frontend/src/components/place/PlaceCheckinModal.vue (hunk 5 above)
Comment (fadykstas) [reply frontend/src/components/place/PlaceCheckinModal.vue]: we need it to highlight previously selected rating

File: routes/api.php [hunk 6]
```diff
@@ -130,6 +130,14 @@
 
         Route::post('/places/rating', 'Api\Place\PlaceRatingController@setRating')
             ->name('place.rating.setPlaceRating');
+      
+        Route::get('/places/rating/byPlaceUser', 'Api\Place\PlaceRatingController@getRating')
```
Comment (nemoipaha) [inline routes/api.php]: @getRating method is duplicated

File: app/Http/Controllers/Api/Place/PlaceController.php [hunk 7]
```diff
@@ -46,25 +51,35 @@ public function __construct(
...
     {
         try {
             $placeResponse = $this->getPlaceItemAction->execute(new GetPlaceItemRequest($id));
+            $placeRatingResponse = $this->getPlaceRatingAction->execute(
+                new GetPlaceRatingRequest($id)
+            );
         } catch (PlaceDoesNotExistException $e) {
             return $this->errorResponse($e->getMessage(), $e->getCode());
+        } catch (PlaceRatingNotFoundException $e) {
+            $placeRatingResponse = new GetPlaceRatingResponse(0, 0, 0, 0);
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Place/PlaceController.php]: not very good to make it like this.
$placeRatingResponse = null;

File: app/Actions/Place/GetPlaceItem/GetPlaceItemPresenter.php [hunk 8]
```diff
@@ -49,7 +50,7 @@ public function __construct(
         $this->photoPresenter = $photoPresenter;
     }
 
-    public function present(GetPlaceItemResponse $placeResponse): array
+    public function present(GetPlaceItemResponse $placeResponse, GetPlaceRatingResponse $placeRatingResponse): array
```
Comment (nemoipaha) [inline app/Actions/Place/GetPlaceItem/GetPlaceItemPresenter.php]: no, you can create another method `presentWithUserRating(GetPlaceItemResponse, ?float $rating)`

File: app/Http/Controllers/Api/Place/PlaceController.php [hunk 9]
```diff
@@ -46,25 +51,35 @@ public function __construct(
...
         }
 
-        return $this->successResponse($presenter->present($placeResponse));
+        return $this->successResponse($presenter->present(
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Place/PlaceController.php]: $presenter->presentWithRating()

File: routes/api.php [hunk 10]
```diff
@@ -130,6 +130,14 @@
...
+            ->name('place.rating.getPlaceRatingByPlaceUser');
+
+        Route::get('/places/rating/{id}', 'Api\Place\PlaceRatingController@getRating')
```
Comment (nemoipaha) [inline routes/api.php]: i think we can remove this route, unused

**Pull request merged**